core/src/main/java/hudson/model/Actionable.java

```diff
@@ -116,7 +130,11 @@
      * @see #getAction(Class)
      */
     public <T extends Action> List<T> getActions(Class<T> type) {
-        return Util.filter(getAllActions(), type);
+        List<T> _actions = Util.filter(getActions(), type);
+        for (TransientActionFactory<?> taf : TransientActionFactory.factoriesFor(getClass(), type)) {
+            _actions.addAll(Util.filter(createFor(taf), type));
```

Comment:
Why are we adding Actions from TransientActionFactories to the result of getActions?